### DISCUSSION TITLE
using vcpkg with VS Code

### DISCUSSION THREAD

**Discussion by siddhsql** [microsoft/vcpkg#40857] [Q&A]
I am trying to use vcpkg with VS Code. From [here](https://learn.microsoft.com/en-us/vcpkg/consume/manifest-mode?tabs=msbuild%2Cbuild-MSBuild):

>You only need to run the vcpkg integrate install command the first time you want to enable MSBuild integration. This enables MSBuild integration for all your existing and future projects. Use vcpkg integrate remove to remove MSBuild system-wide integration.

>This integration method automatically adds vcpkg-installed packages to the following project properties: Include Directories, Link Directories, and Link Libraries.  Additionally, this creates a post-build action that ensures that any required DLLs are copied into the build output folder.  This works for all solutions and projects using Visual Studio 2015 or newer.

How can I get this but for VS Code? My workflow is that I use VS Code as the IDE and I build my projects from the command line by running `msbuild` myself. I am even authoring `.vcxproj` files myself with some help from CMake. I want to know what changes I need to make to my `vcxproj` file so that:

1. I get vcpkg's Include Directories, Link Directories, and Link Libraries. 
2. create a post-build action that ensures that any required DLLs are copied into the build output folder.

**Comment by AugP** [comment]
If you're using CMake anyway, you can just use CMake natively in VS Code, which should be easier. There is a CMake Tools extension for VS Code as well. We also have a tutorial for using vcpkg with CMake in VS Code: https://learn.microsoft.com/vcpkg/get_started/get-started-vscode. Generating MSBuild project files with CMake and hand editing them to add vcpkg integration is more pain. If you're going to use MSBuild it would be easier to use the Visual Studio IDE, which has native integration for MSBuild and a GUI experience for editing those projects.
